### ISSUE TITLE
Feature Request: Support GoRedis

### ISSUE THREAD

**Issue by ryanbyyc** [gocraft/work#6]
https://github.com/go-redis/redis

This is a personal request as I prefer GoRedis over redigo. For the time being I have the two connection clients.

**Comment by cypriss** [comment]
Fair enough. If you know of a way to abstract the redis client I'd be all ears. I don't know of an easy way to do it off the top of my head.

**Comment by chambo-e** [comment]
I was trying to abstract redis as an interface but I came across some troubles to simulate similar behaviour in redigo and goredis.
The main one is the way redigo handle pipelined commands, it has a really nice way to buffer commands and then receive replies one by one that's heavily used in `gocraft/work` while goredis wait for every response before returning the pipe (or I did not find how). 

This would affect `gocraft/work` performances thus I don't really know if it's worth it to support other redis drivers
